### PR TITLE
OLS-624: Check attachments metadata

### PR DESCRIPTION
**Pull request by tisnik**
## Description

- Check attachments metadata
- Unit tests to cover new functions
- e2e tests to cover new functionality on REST API level

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library
- [ ] Bump-up library or tool used for development (does not change the final image)
- [x] Unit tests
- [x] End to end tests

## Related Tickets & Documents

- Related Issue #[OLS-624](https://issues.redhat.com//browse/OLS-624)
- Closes #

### CODE REVIEW

File: ols/app/endpoints/ols.py [hunk 1]
```diff
@@ -84,6 +85,9 @@ def conversation_request(
     conversation_id = retrieve_conversation_id(llm_request)
     previous_input = retrieve_previous_input(user_id, llm_request)
 
+    # Retrieve attachments from the request
+    retrieve_attachments(llm_request)
```
[the remark this answers is not in the document]
Comment (tisnik) [reply ols/app/endpoints/ols.py:89]: part of follow up. If I add `attachments = retrieve_attachments(llm_request)` linter will be unhappy with unused variable

File: ols/constants.py [hunk 2]
```diff
@@ -160,3 +160,11 @@ class GenericLLMParameters:
 RUNNING_IN_CLUSTER = (
     "KUBERNETES_SERVICE_HOST" in os.environ and "KUBERNETES_SERVICE_PORT" in os.environ
 )
+
+# Supported attachment types
+ATTACHMENT_TYPES = frozenset({"log", "configuration", "error message", "stack trace"})
```
[the remark this answers is not in the document]
Comment (xrajesh) [reply ols/constants.py:165]: I was thinking `configuration` is `api object` . Also , would like to consider events , the one we see in the OCP console dashboards . I am okay if we use `api object` and send as yaml files.

[the remark this answers is not in the document]
Comment (tisnik) [reply ols/constants.py]: updated by adding the `"api object"` there

**Pull request merged**